dbt_subprojects/dex/macros/models/tapio_liquidity_and_volume.sql

```diff
@@ -0,0 +1,167 @@
+        SELECT 
+            date_trunc('day', e.block_time) AS day,
+            e.pool_address,
+            p.token0_address,
+            p.token1_address,
+            t0.symbol AS token0_symbol,
+            t1.symbol AS token1_symbol,
+            SUM(CASE 
+                WHEN e.event_type IN ('Minted', 'Donated') AND e.amount IS NOT NULL THEN CAST(e.amount AS DOUBLE)
+                ELSE 0 
+            END) AS token0_balance_raw,
+            SUM(CASE 
+                WHEN e.event_type IN ('Redeemed', 'TokenSwapped') AND e.amount IS NOT NULL THEN CAST(e.amount AS DOUBLE)
+                ELSE 0 
+            END) AS token1_balance_raw
+        FROM unified_events e
+        JOIN {{ source(blockchain, pools_table) }} p ON e.pool_address = p.address
+        JOIN {{ source(blockchain, tokens_table) }} t0 ON p.token0_address = t0.address
+        JOIN {{ source(blockchain, tokens_table) }} t1 ON p.token1_address = t1.address
+        GROUP BY 1, 2, 3, 4, 5, 6
+    )
+
+    running_balance AS (
+        -- Fill missing days to maintain continuity
+        SELECT 
+            ds.day,
+            ed.pool_address,
+            ed.token0_address,
+            ed.token1_address,
+            ed.token0_symbol,
+            ed.token1_symbol,
+            COALESCE(ed.token0_balance_raw, 0) AS token0_balance_raw,
+            COALESCE(ed.token1_balance_raw, 0) AS token1_balance_raw
+        FROM (
+            SELECT generate_series(
+                COALESCE((SELECT MIN(day) FROM event_data), CURRENT_DATE - INTERVAL '30 days'),
+                COALESCE((SELECT MAX(day) FROM event_data), CURRENT_DATE),
+                INTERVAL '1' day
+            ) AS day
+        ) ds
+        LEFT JOIN event_data ed ON ds.day = ed.day
+    ),
+
+    final_data AS (
+        -- Convert balances to scaled values and USD prices
+        SELECT 
+            rb.day,
+            rb.pool_address,
+            rb.token0_address,
+            rb.token1_address,
+            rb.token0_symbol,
+            rb.token1_symbol,
+            rb.token0_balance_raw / POWER(10, COALESCE(t0.decimals, 18)) AS token0_balance_scaled, 
+            rb.token1_balance_raw / POWER(10, COALESCE(t1.decimals, 18)) AS token1_balance_scaled,
+            (rb.token0_balance_raw / POWER(10, COALESCE(t0.decimals, 18))) * p0.usd_price AS token0_liquidity_usd,
+            (rb.token1_balance_raw / POWER(10, COALESCE(t1.decimals, 18))) * p1.usd_price AS token1_liquidity_usd
+        FROM running_balance rb
+        JOIN {{ source(blockchain, tokens_table) }} t0 ON rb.token0_address = t0.address
+        JOIN {{ source(blockchain, tokens_table) }} t1 ON rb.token1_address = t1.address
+        LEFT JOIN {{ source(blockchain, prices_table) }} p0 ON rb.token0_address = p0.token_address AND rb.day = p0.day
+        LEFT JOIN {{ source(blockchain, prices_table) }} p1 ON rb.token1_address = p1.token_address AND rb.day = p1.day
+    ),
+
+    volume_data AS (
```

Comment:
would be great to have a daily metrics view but I suggest we keep it separate for now. this should be the `liquidity` (only) spell and in the future we can merge in another spell the main metrics: fees, liquidity, volume.